### ISSUE TITLE
A problem about transform code

### ISSUE THREAD

**Issue by Gaoyiminggithub** [jfzhang95/pytorch-deeplab-xception#20]
Hi,

Thanks for the great library, i see the code custom_transforms.py, in class ToTensor , `mask[mask == 255] = 0` (line 127).
why need to set mask == 255 to 0 ? 255 means ignore index which should not be set to background. Am i right?

**Comment by jfzhang95** [comment]
Yes you're right, label == 255 should be ignored when computing IoU and Loss.
I set label == 255 to 0 because I think these areas are also background. 
You can skip it if you want~

**Comment by FJR-Nancy** [comment]
Label == 255 could not be simply seen as background, because it will influence the results when computing IoU and Loss. The locations with gt label == 255 should be ignored, but after being set to 0, they are not ignored but are considered while computing background IoU and loss. I tried to remove this code, and got mIoU on val from 66.191% to 66.753%.

**Comment by jfzhang95** [comment]
@FJR-Nancy I know set label == 255 to 0 will effect the results. But I did not know whether the influence is good or not when set boundary areas (label == 255) to background. Thank you for your reports. I will remove this line.

**Comment by FJR-Nancy** [comment]
By the way, the transform has another problem that the random scaling does not really work. In RandomSized(), the image and label are resized to a random scale, however, in Scale() and RandomCrop(), they are resized back to the needed size. Here,  RandomCrop() does not really work, either, because before crop(), image and label have been resized to the needed size already. 
If you look at the official code of DeepLab v3+ in TensorFlow [https://github.com/tensorflow/models/blob/master/research/deeplab/input_preprocess.py](url) , you will see Pad() and RandomCrop() are needed to support RandomScale().

**Comment by jfzhang95** [comment]
OK, I will look at it closely. Thank you for pointing this out.

**Comment by Gaoyiminggithub** [comment]
@FJR-Nancy Hi, have you tried other learning rate, such as 0.007 ? When i tried other larger learning rate, the miou is worse than the lr set to 1e-7. Do you know why...

**Comment by FJR-Nancy** [comment]
@Gaoyiminggithub Yes, I have tried, before which I removed “size_average=False” in cross_entropy2d(). Because the loss value is high since the "size_average" is set to False while computing the loss. You can set it to True or just remove it -- which will use “elementwise_mean” by default in PyTorch. Then you could increase the initial learning rate accordingly, maybe 0.007, which is the same as the paper. I got 66.191% after doing this compared to 65.958％.
